WalletWasabi/Blockchain/TransactionBuilding/BnB/SelectionStrategy.cs

```diff
@@ -1,52 +1,54 @@
-using System.Diagnostics;
 using System.Linq;
 
 namespace WalletWasabi.Blockchain.TransactionBuilding.BnB;
 
 public abstract class SelectionStrategy
 {
-	/// <param name="target">Value in satoshis.</param>
-	/// <param name="inputValues">Values in satoshis of the coins the user has (in descending order).</param>
-	/// <param name="inputCosts">Costs of spending coins in satoshis.</param>
-	public SelectionStrategy(long target, long[] inputValues, long[] inputCosts, CoinSelection bestSelection)
+	/// <param name="parameters">Parameters of the strategy specifying input coins, the target and final selection restrictions.</param>
+	/// <param name="bestSelection">Best selection so far.</param>
+	public SelectionStrategy(StrategyParameters parameters, CoinSelection bestSelection)
 	{
-		InputCosts = inputCosts;
-		InputValues = inputValues;
-		Target = target;
+		Parameters = parameters;
 		BestSelection = bestSelection;
 
-		RemainingAmounts = new long[inputValues.Length];
+		RemainingAmounts = new long[InputValues.Length];
```

Comment:
Nit:
Shouldn't we use the local `parameters` for these? Like: `new long[parameters.InputValues.Length];`

Reply:
No, no, the change is not necessary. I was just curious.